pkg/block/fetcher.go

```diff
@@ -484,9 +495,26 @@ func (f *DeduplicateFilter) filterForResolution(root *Node, metaSlice []*metadat
 }
 
 // DuplicateIDs returns slice of block ids
-// that are filtered out by DeduplicateFilter.
-func (f *DeduplicateFilter) DuplicateIDs() []ulid.ULID {
-	return f.duplicateIDs
+// that are filtered out by DeduplicateFilter and not marked to be deleted.
+func (f *DeduplicateFilter) DuplicateIDs() ([]ulid.ULID, error) {
+	toFilterIds := []ulid.ULID{}
+
+	deletionMarkBlocks := f.ignoreDeletionMarkFilter.DeletionMarkBlocks()
```

Comment:
Let's comment on why we do this! (: 

Also... does compactor has to have any ignore delay? :thinking: Maybe only readers needs :thinking: I think on compactor we are fine.

Reply:
Maybe compactor can have 0 ignore delay so we don't this logic.

Reply:
No, this is especially required for compact, since compact is also reading metas.
This is required for the case that when we will compact the metas, we don't want them to come up in the next compaction cycle. 

The `ignoreDeletionMarkFilter` only filters the blocks which have been marked for deletion `deleteDelay/2` duration ago. So the blocks which were compacted in last cycle will not be filtered out by `ignoreDeletionMarkFilter`, thus these blocks will be considered in the next compaction cycle.

In order to address that, we are using `ignoreDeletionMarkFilter` in `DeduplicateFilter` so that the blocks which were marked for deletion in the last compaction cycle aren't again considered for this compaction cycle.

Reply:
I am not passing `ignoreDeletionMarkFilter` in `DeduplicateFilter` now.
I am using both of them in `compact.GarbageCollect` function call. I agree that DeduplicateFilter should not be able to access `ignoreDeletionMarkFilter`

Reply:
Resolving this since the latest feedback doesn't contain improvement points on this.